src/Symfony/Component/Intl/Locale.php

```diff
@@ -43,7 +43,7 @@ public static function setDefaultFallback($locale)
     /**
      * Returns the default fallback locale.
      *
-     * @return string The default fallback locale
+     * @return string|null The default fallback locale
```

Comment:
Looks like a BC break. Before, you were guaranteed to get `en` as a fallback.

Reply:
it's a backport of 4.2 https://github.com/symfony/symfony/blob/6e8529f82ec4c7ddb99c6330c0443ed4e0efbb48/src/Symfony/Component/Intl/Locale.php#L34 :) for consistency across branches.

you could set a `null` fallback locale as of 3.4 (now with test)

Reply:
the snippet you mention is about `getFallback()`, not `getDefaultFallback()`. In 3.4, how/when can you get a `null`default fallback?

Reply:
https://github.com/symfony/symfony/blob/6c6f76f217ca200149ae1630799a43ce2bc27545/src/Symfony/Component/Intl/Locale.php#L80

currently we rely on it during compilation (as of 3.4)

https://github.com/symfony/symfony/blob/b2df5ca2b2d4818731836c545e83f94e7d3f391a/src/Symfony/Component/Intl/Resources/bin/update-data.php#L196

the possibilities are

"null": from "nl" fallback to null
"en": from "nl" fallback to "en", then "root", then "null"
"root": from "nl" fallback ti "root", then "null"

besides, this class is internal ;)
this PR changes it troot, but having "no fallback" (no "en", nor "root")

Reply:
given this class is internal, we could favor a string api, so the root fallback is always involved. But it's a different topic, and perhaps being able to exclude "root" / any fallback by using "null" is a nice to have/keep ...

Reply:
The fact that this class being internal is irrelevant, people are using the class and the methods.